scipy/stats/_continuous_distns.py

```diff
@@ -6973,6 +6973,171 @@ def _stats(self, a):
 
     def _entropy(self, a):
         return 1 - 1.0/a - np.log(a)
+    
+    def _support_mask(self, x, a):
+        if np.any(a < 1):
+            return (x != 0) & super(powerlaw_gen, self)._support_mask(x, a)
+        else:
+            return super(powerlaw_gen, self)._support_mask(x, a)
+
+    def fit(self, data, *args, **kwds):
+        '''
+        The strategy for this fit method is complex, so its steps will be
+        described here first. The following steps occur after validation
+        of fixed parameters and data.
+
+        1) If the scale and location are fixed, return the analytically
+           determined shape.
+
+        2) If the scale is fixed, we find a fit with the analytical equations
+           that correspond to the shape being less than and greater than one,
+           and then return whichever is better (according to the log-likelihood
+           function).
+
+        3) If the location is fixed, return the analytically determined scale
+           and shape (or if the shape is fixed, the fixed shape. At this point
+           the scale is known to be free.
+
+        4) If the location, scale, and shape are all free, we find the
```

Comment:
I am not sure if I understand this argument: couldn't there be a solution with a > 1 that will produce a greater likelihood?

Reply:
This assumes there is only one local optimum, and that is the global optimum. @swallan did we find a counter-example? I vaguely remember at point trying the logic for both a < 1 and a > 1 then picking the better of the two results.

Reply:
Yes, we ran across the need to do that in case 2 (scale fixed), so it makes sense to extend the idea to the case where scale is not fixed.